### PR TITLE
fix(ui): fixed table threshold bug and added option to make thresholds populate text or background

### LINKED ISSUE #16309: Table thresholding should be deactivated my default
Screenshot 2019-12-20 11 19 27

By default no thresholds or base colors should be applied until a threshold has been added. Alternatively the base color should be the same as the tables base color.

### LINKED ISSUE #16359: Add the ability for the user to choose text or background for thresholds
similar to 1.x, whenever a user defines a threshold in a graph, allow them to choose between changing the color of the text or changing the color of the background. similar to 1.x chronograf.

### PR DESCRIPTION
**Pull request by Asalem1**
Closes #16309
Closes #16359

### Problem

Table thresholds were automatically populating the background color rather than giving users an option.

### Solution

Added option to toggle between views for table threshold so user can set background or text color. Changed the default table threshold base color to white after speaking with @alexpaxton about design and the need for a default base color.

- [ ] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)

### CODE REVIEW

File: ui/src/shared/components/tables/TableCell.tsx [hunk 1]
```diff
@@ -42,7 +42,7 @@ class TableCell extends PureComponent<Props> {
           onClick={this.handleClick}
           data-column-index={columnIndex}
           data-row-index={rowIndex}
-          data-testID={`${data}-table-header`}
+          data-testid={`${data}-table-header`}
```
Comment (Asalem1) [inline ui/src/shared/components/tables/TableCell.tsx]: this was to fix a lingering linter error

File: ui/src/shared/components/ThresholdSetting.tsx [hunk 2]
```diff
@@ -57,7 +57,7 @@ const ThresholdSetting: FunctionComponent<Props> = ({
   } else if (type === COLOR_TYPE_MAX) {
     label = 'Maximum'
   } else {
-    label = 'Value is <='
+    label = 'Value is >='
```
Comment (Asalem1) [inline ui/src/shared/components/ThresholdSetting.tsx]: this change more accurately reflects the functionality of setting the thresholds

File: ui/src/shared/constants/thresholds.ts [hunk 3]
```diff
@@ -119,3 +119,13 @@ export const DEFAULT_THRESHOLDS_LIST_COLORS = [
     value: 0,
   },
 ]
+
+export const DEFAULT_THRESHOLDS_TABLE_COLORS = [
```
Comment (Asalem1) [inline ui/src/shared/constants/thresholds.ts]: updated the base table color based on discussions with @alexpaxton on what would look best

File: ui/src/shared/constants/colorOperations.ts [hunk 4]
```diff
@@ -88,10 +85,6 @@ export const generateThresholdsListHexs = ({
       lastValueNumber
     )
 
-    if (cellType === 'table') {
```
Comment (Asalem1) [inline ui/src/shared/constants/colorOperations.ts]: removing these lines reverted the functionality to its original format, restoring it to what it was originally supposed to do

File: ui/src/timeMachine/components/view_options/TableOptions.tsx [hunk 5]
```diff
@@ -131,6 +132,9 @@ export class TableOptions extends Component<Props, {}> {
             onSetThresholds={onSetColors}
           />
         </Grid.Column>
+        <Grid.Column>
```
Comment (Asalem1) [inline ui/src/timeMachine/components/view_options/TableOptions.tsx]: adds a toggle to allow users to switch between threshold colors being set to the background or text color

File: ui/src/shared/components/tables/TableCell.tsx (hunk 1 above)
Comment (drdelambre) [reply ui/src/shared/components/tables/TableCell.tsx]: 🤦‍♂

**Comment by drdelambre** [comment]
lgtm

**Pull request merged**